service/matching/matchingEngine.go

```diff
@@ -751,22 +751,33 @@ func (e *matchingEngineImpl) InvalidateTaskQueueMetadata(
 ) (*matchingservice.InvalidateTaskQueueMetadataResponse, error) {
 	namespaceID := namespace.ID(req.GetNamespaceId())
 	taskQueueName := req.GetTaskQueue()
-	taskQueue, err := newTaskQueueID(namespaceID, taskQueueName, enumspb.TASK_QUEUE_TYPE_WORKFLOW)
-	if err != nil {
-		return nil, err
-	}
-	tqMgr, err := e.getTaskQueueManager(hCtx, taskQueue, enumspb.TASK_QUEUE_KIND_NORMAL, false)
-	if tqMgr == nil && err == nil {
-		// Nothing to do here
-		return &matchingservice.InvalidateTaskQueueMetadataResponse{}, nil
-	}
-	if err != nil {
-		return nil, err
-	}
-	err = tqMgr.InvalidateMetadata(req)
-	if err != nil {
-		return nil, err
+
+	// Invalidation matters for everything that isn't the root partition workflow type, since they are caching data
+	// which they fetch from that queue.
+	for _, tqt := range []enumspb.TaskQueueType{enumspb.TASK_QUEUE_TYPE_WORKFLOW, enumspb.TASK_QUEUE_TYPE_ACTIVITY} {
```

Comment:
I'm actually kinda having second thoughts on the decision to not include tqt in the protos here. It sorta makes sense but then it means you have to do stuff like this. I'm open to revisiting that